src/hooks/useSetSentry.ts

```diff
@@ -0,0 +1,40 @@
+import { setContext, setUser } from "@sentry/nextjs"
+import { useWalletSelector } from "@src/providers/WalletSelectorProvider"
+import { useEffect } from "react"
+import { useAccount } from "wagmi"
+
+export const useSentrySetUser = ({ userAddress }: { userAddress?: string }) => {
+  useEffect(() => {
+    if (userAddress) {
+      setUser({ id: userAddress })
+    }
+    return () => {
+      setUser(null)
+    }
```

Comment:
I'm very sorry about lot of comments for such small PR :) But it should be `else` statement.